### PR TITLE
feat: add pages and page blocks

### PR DESCRIPTION
**Pull request by iamsahebgiri**
- Created the page UI as per design.
- Implemented add page to favorite and remove page from favorite.
- Can able to update and delete page
- Can able to create, update and delete page blocks.

### CODE REVIEW

File: apps/app/components/pages/create-update-page-modal.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+  const { workspaceSlug, projectId } = router.query;
+
+  const { setToastAlert } = useToast();
+
+  const onClose = () => {
+    handleClose();
+  };
+
+  const createPage = async (payload: IPageForm) => {
+    payload = {
```
Comment (aaryan610) [inline apps/app/components/pages/create-update-page-modal.tsx:37]: Why do we need this line?

File: apps/app/components/pages/delete-page-modal.tsx [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+  const handleClose = () => {
+    setIsOpen(false);
+    setIsDeleteLoading(false);
+  };
+
+  const handleDeletion = async () => {
+    setIsDeleteLoading(true);
+    if (!data || !workspaceSlug) return;
+    console.log(data);
```
Comment (aaryan610) [inline apps/app/components/pages/delete-page-modal.tsx:46]: Remove this console.log

File: apps/app/components/pages/delete-page-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        handleClose();
+
+        setToastAlert({
+          title: "Success",
+          type: "success",
+          message: "Page deleted successfully",
+        });
+      })
+      .catch((error) => {
+        console.log(error);
```
Comment (aaryan610) [inline apps/app/components/pages/delete-page-modal.tsx:64]: Add a toast alert to handle this error

File: apps/app/pages/[workspaceSlug]/projects/[projectId]/pages/[pageId].tsx [hunk 4]
```diff
@@ -0,0 +1,227 @@
...
+
+const PageBlock: React.FC<any> = ({ pageBlock }: { pageBlock: IPageBlock }) => {
+  const [name, setName] = useState(pageBlock.name);
+  const [showButtons, setShowButtons] = useState(false);
+  const { setToastAlert } = useToast();
+  const {
+    query: { workspaceSlug, projectId, pageId },
+  } = useRouter();
+
+  const handleMouseEnter = () => {
```
Comment (aaryan610) [inline apps/app/pages/[workspaceSlug]/projects/[projectId]/pages/[pageId].tsx:37]: Use tailwind group to handle the hover events.

File: apps/app/services/pages.service.ts [hunk 5]
```diff
@@ -0,0 +1,158 @@
...
+      .catch((error) => {
+        throw error?.response?.data;
+      });
+  }
+
+  async patchPage(
+    workspaceSlug: string,
+    projectId: string,
+    pageId: string,
+    data: IPageForm
```
Comment (aaryan610) [inline apps/app/services/pages.service.ts:25]: data needs to be Partial<IPageForm> for the patch request

File: apps/app/pages/[workspaceSlug]/projects/[projectId]/pages/[pageId].tsx [hunk 6]
```diff
@@ -0,0 +1,200 @@
...
+            updatePageBlock();
+          }
+        }}
+        onChange={(e) => {
+          setName(e.target.value);
+        }}
+        className="border-none bg-transparent outline-none"
+      />
+
+      <div className="absolute top-[50%] right-0 z-20 mr-2 hidden translate-y-[-50%] group-hover:block">
```
Comment (aaryan610) [inline apps/app/pages/[workspaceSlug]/projects/[projectId]/pages/[pageId].tsx:100]: We don't need to use absolute positioning, just flex justify between would be fine

Comment (iamsahebgiri) [reply apps/app/pages/[workspaceSlug]/projects/[projectId]/pages/[pageId].tsx]: yeah I have fixed it

**Review by aaryan610 — APPROVED** [review]

**Pull request merged**